be/src/exec/pipeline/exchange/exchange_sink_operator.cpp

```diff
@@ -316,22 +312,22 @@ Status ExchangeSinkOperator::push_chunk(RuntimeState* state, const vectorized::C
     if (num_rows == 0) {
         return Status::OK();
     }
+
     if (_part_type == TPartitionType::UNPARTITIONED || _channels.size() == 1) {
-        // We use sender request to avoid serialize chunk many times.
-        // 1. create a new chunk PB to serialize
-        ChunkPB* pchunk = _chunk_request.add_chunks();
+        // 1. create a PTransmitChunkParams to serialize chunk as common datas.
+        PTransmitChunkParams request;
+        ChunkPB* pchunk = request.add_chunks();
+
         // 2. serialize input chunk to pchunk
         RETURN_IF_ERROR(serialize_chunk(chunk.get(), pchunk, &_is_first_chunk, _channels.size()));
         _current_request_bytes += pchunk->data().size();
+
         // 3. if request bytes exceede the threshold, send current request
         if (_current_request_bytes > _request_bytes_threshold) {
-            butil::IOBuf attachment;
-            // construct_brpc_attachment(&_chunk_request, &attachment);
```

Comment:
The brpc protobuf protocol header has a size limit. What happens when the request body is very large if we don't use attachments

Reply:
I've supplement attachment at broadcast and shuffle.